atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/sweep/asts/DefaultCandidateSweepableBucketRetriever.java

```diff
@@ -0,0 +1,179 @@
+    // But we've not preconditioning on it since zero is acceptable for testing.
+    private final Refreshable<Duration> minimumDurationBetweenRefresh;
+    // We use a supplier for the jitter (rather than directly using ThreadLocalRandom) to make testing simpler.
+    private final Supplier<Long> jitterMillisGenerator;
+
+    // Passed in for testing
+    private final Clock clock;
+
+    // We use a jitter to prevent thundering herd issues at the macro scale. The underlying SweepableBucketRetriever
+    // should have jitter between requests, but we also want to have the batch requests be scheduled at different times
+    // across nodes.
+
+    // This is a field because we want to have a constant jitter for the duration of the refresh, rather than
+    // having each requested update use a different jitter and having an inconsistent debounce window.
+    private volatile long jitterMillis;
+
+    @VisibleForTesting
+    DefaultCandidateSweepableBucketRetriever(
+            SweepableBucketRetriever sweepableBucketRetriever,
+            Refreshable<Duration> minimumDurationBetweenRefresh,
+            Clock clock,
+            Supplier<Long> jitterMillisGenerator) {
+        this.sweepableBucketRetriever = sweepableBucketRetriever;
+        this.minimumDurationBetweenRefresh = minimumDurationBetweenRefresh;
+        this.clock = clock;
+
+        // Coalescing Supplier doesn't have the exact semantics we want - for requests that come in after the
+        // computation has started, they will trigger a new computation after the first one is complete. However,
+        // we would like to have the request to be a no-op if there's already one in progress (rather than trigger
+        // a new request).
+        // Despite that, it's simpler to use the battle tested coalescing supplier, and mitigate this with the already
+        // needed minimumDurationBetweenRefresh, since the next iteration will happen straight after
+        // which should still be within the debounce window.
+        this.supplier = new CoalescingSupplier<>(this::getNext);
+        this.jitterMillisGenerator = jitterMillisGenerator;
+        this.jitterMillis = jitterMillisGenerator.get();
+    }
+
+    public static CandidateSweepableBucketRetriever create(
+            SweepableBucketRetriever sweepableBucketRetriever,
+            Refreshable<Duration> debouncerDuration,
+            Refreshable<Duration> maxJitter,
+            Clock clock) {
+        return new DefaultCandidateSweepableBucketRetriever(sweepableBucketRetriever, debouncerDuration, clock, () -> {
+            // This is not full jitter. I chose to require a minimum delay, because we simply _do not need_ to refresh
+            // buckets that quickly, and the scans over the database aren't precisely free.
+            // This jitter only really matters for things calling requestUpdate frequently (as opposed to the background
+            // task, which should have its own jitter)
```

Comment:
The background task does not currently have any jitter, but I'm tracking adding some in another PR.